### PR TITLE
PT-3660: If the DetailsDialog contains no data, "no additional information" should be displayed on loss of focus

### PR DESCRIPTION
**Pull request by a-gvo**

### CODE REVIEW

File: components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml [hunk 1]
```diff
@@ -7166,11 +7177,31 @@ document.observe("xwiki:dom:updated", init);
           var _this = this;
           this._addDetailsButton.observe('click', function (event) {
               event.stop();
-              _this.addDialog(false, _this._globalQualifierCount &amp;&amp; _this._globalQualifierCount())
+              _this._shouldAddDialog.bind(_this)() &amp;&amp; _this.addDialog(false, _this._globalQualifierCount &amp;&amp; _this._globalQualifierCount())
```
Comment (sdumitriu) [inline components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml]: Why `bind` it?

File: components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml [hunk 2]
```diff
@@ -6066,6 +6066,11 @@ document.observe("xwiki:dom:updated", init);
               this._termDetails.removeClassName('focused');
               this._termDetails.addClassName('blurred');
               this._toggleSummarize();
+              if(!this._termDetails.down("dl &gt; *:not([style='display: none;']):not(.hidden)")) {
```
Comment (sdumitriu) [inline components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml]: I'm not a big fan of the `style=` check. Are you sure it works in all browsers, including IE?

File: components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml (hunk 1 above)
Comment (a-gvo) [reply components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml]: Oops. Removed.

File: components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml (hunk 2 above)
Comment (a-gvo) [reply components/widgets/ui/src/main/resources/PhenoTips/Widgets.xml]: Not sure, actually. Replaced it with a different check.

**Pull request merged**